### PR TITLE
Add a LevelDB database impl.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -650,7 +651,7 @@ cpp_log_database_large_test_LDADD = \
 	cpp/libcore.a \
 	cpp/libtest.a \
 	$(libevent_LIBS) \
-	-lglog -lgflags -lprotobuf -lpthread -lcrypto -lsqlite3
+	-lglog -lgflags -lprotobuf -lpthread -lcrypto -lsqlite3 -l leveldb
```
Comment (pphaneuf) [inline Makefile.am]: Spurious space.

File: Makefile.am [hunk 2]
```diff
@@ -155,7 +156,7 @@ cpp_server_ct_server_LDADD = \
 	cpp/libcore.a \
 	$(json_c_LIBS) \
 	$(libevent_LIBS) \
-	-lglog -lgflags -lcrypto -lprotobuf -lpthread -lsqlite3
+	-lglog -lgflags -lcrypto -lprotobuf -lpthread -lsqlite3 -lleveldb
```
Comment (pphaneuf) [inline Makefile.am]: From what I can see on Travis, you'll need another library everywhere you need `-lleveldb`?

File: Makefile.am (hunk 1 above)
Comment (AlCutter) [inline Makefile.am]: Dinged

**Comment by pphaneuf** [comment]
This needs a good rebase on top of `master`.

**Comment by pphaneuf** [comment]
Also need to update it for the duplicate submission support, I think?

```
[ RUN      ] DBTest/2.CreateSequencedDuplicateEntryNewSequenceNumber
cpp/log/database_test.cc:117: Failure
Value of: this->db()->CreateSequencedEntry(duplicate_cert)
  Actual: 0
Expected: DB::ENTRY_ALREADY_LOGGED
Which is: 3
[  FAILED  ] DBTest/2.CreateSequencedDuplicateEntryNewSequenceNumber, where TypeParam = LevelDB<cert_trans::LoggedCertificate> (8 ms)
```

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**